public/app/plugins/datasource/grafana-azure-monitor-datasource/components/AnalyticsConfig.tsx

```diff
@@ -91,6 +91,18 @@ export const AnalyticsConfig: FunctionComponent<Props> = (props: Props) => {
     });
   };
 
+  const onClearAzLogsCreds = () => {
+    updateOptions((options) => {
+      return {
+        ...options,
+        jsonData: {
+          ...options.jsonData,
+          azureLogAnalyticsSameAs: true,
+        },
+      };
```

Comment:
Lets clean up the object properly and `delete` the deprecated fields `azureLogAnalyticsSameAs`, `logAnalyticsTenantId`, `logAnalyticsClientId`, and `logAnalyticsSubscriptionId`.

The goal here should be to get their JSON data into a close as "ideal" or "as if configured from scratch" state as possible, as they're doing a manual action.

Reply:
I would move it to `credentials.ts`, I wouldn't do data manipulation in the UI component.

Reply:
my reasoning to not delete those were in case  some emergency recovery is needed but it's probably not the case.